### DISCUSSION TITLE
Dynamic Routes w/ special chars best practices

### DISCUSSION THREAD

**Discussion by jamezening** [vercel/next.js#37565] [Help]
Hi,

I have a site that uses Dynamic Routes, which can contain all and any characters.

I am having some trouble when these routes contain certain characters, so when linking to a Dynamic route, I encode the dynamic route path using `encodeURIComponent`.

This helps the issue a bit, but another issue arises when using Static Generation to pre-build dynamic routes with tricky/special characters.

[This Mozilla link](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent) has an example function that "sanitizes" other potential characters not converted by `encodeURIComponent`:

```
function fixedEncodeURIComponent(str) {
  return encodeURIComponent(str).replace(/[!'()*]/g, function(c) {
    return '%' + c.charCodeAt(0).toString(16).toUpperCase();
  });
}
```
, but it would be nice to know if there are any best practices/guides for safely clearing all special characters in Dynamic Routes.

Any help appreciated!

**Comment by Crinchy** [comment]
Hey, did you found any fix for it?
